devel/yarn/Portfile

```diff
@@ -29,17 +29,8 @@ checksums           rmd160  be20072bfd66a6411911bebb63284abb7f7f454c \
                     sha256  883df435f68ce47c93c2e4be27acbd4122ae52ef4d334f75104c0c5e187a9173
 
 depends_run         bin:node:nodejs6
-variant nodejs4 conflicts nodejs6 nodejs7 description {Use Node.js v4} {
-    depends_lib-replace bin:node:nodejs6 port:nodejs4
-}
-variant nodejs6 conflicts nodejs4 nodejs7 description {Use Node.js v6} {
-    depends_lib-replace bin:node:nodejs6 port:nodejs6
-}
-variant nodejs7 conflicts nodejs4 nodejs6 description {Use Node.js v7} {
-    depends_lib-replace bin:node:nodejs6 port:nodejs7
-}
 
-post-patch {
+patch {
```

Comment:
I believe that you misunderstood what @larryv wanted to tell you. Not that this should be changed from `post-patch` into `patch` (you should probably never do that), but you should create a proper patch file, perhaps call it patch-bin-yarn.diff, put it under `files/` and add a line
```
patchfiles patch-bin-yarn.diff
```
to the Portfile.

Regular expressions are too fragile, in particular when upgrading to a newer versions and this one is definitely too complex. The `reinplace` is only used when you need to substitute `${prefix}` or so.

In fact I believe that this is what you want. I don't want to decrypt or guess what your expression tries to do, but you can replace the whole line with
```
basedir=@@PREFIX@@
```
and then do
```
reinplace "s|@@PREFIX@@|${prefix}|g"
```
I'm not sure if that's actually the proper path, but I guess it might be something in that direction.